src/python/pants/util/contextutil.py

```diff
@@ -80,13 +80,13 @@ def _purge_env():
 
 def _restore_env(env):
   for k, v in env.items():
-    os.environ[k] = _os_encode(v)
+    os.environ[k] = v if PY3 else _os_encode(v)
 
 
 @contextmanager
 def hermetic_environment_as(**kwargs):
   """Set the environment to the supplied values from an empty state."""
-  old_environment = _copy_and_decode_env(os.environ)
+  old_environment = os.environ.copy() if PY3 else _copy_and_decode_env(os.environ)
```

Comment:
It seems like it would be cleaner to do the PY3 check in `_copy_and_decode_env` and `_os_encode` rather than before calling those methods?